client/src/components/profile.ts

```diff
@@ -193,13 +231,13 @@ export class ProfileConfig {
    * @param name {@link String} of the profile name
    * @returns Profile object
    */
-  getProfileByName(name: string): Profile {
-    let profile: Profile = undefined;
+  getProfileByName<T extends Profile>(name: string): T {
     const profileList = this.getAllProfiles();
     if (name in profileList) {
-      profile = profileList[name];
+      /* eslint-disable @typescript-eslint/consistent-type-assertions*/
```

Comment:
After fighting with the type system, I resorted to asserting the profile as the requested T type here. Open to alternative suggestions on how we might let the caller specify the specific subtype returned when fetching the profile.

Reply:
I'm going to opt to leave this as is if agreeable.